src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  color: Color,
+  shape: Shape,
 
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((x, y) => y - x);
 
-export class Circle {
+    if ((sides[0] >= sides[1] + sides[2])
+        || a < 1 || b < 1 || c < 1) {
+      throw new Error('All sides must be greater than 0');
+    }
+  }
 
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiPerimeter: number = (a + b + c) / 2;
+
+    return Number(
+      (Math.sqrt(semiPerimeter
+        * (semiPerimeter - a)
+        * (semiPerimeter - b)
+        * (semiPerimeter - c))
+      ).toFixed(2),
+    );
```

Comment:
i think rounding helper function will be helpful as we have to round up all three  'getArea' values.